### DISCUSSION TITLE
Use a feature braching strategy

### DISCUSSION THREAD

**Discussion by runkelcorey** [canimus/cuallee#317] [Ideas]
I'm using cuallee for several production jobs. I love the lightweight interface but faced issues in several jobs this week while trying to run cuallee on Python 3.9. The fix was easy but I wish there were a way to avoid this issue. I suggest:
- branch protections on `main` preventing pushes without PRs
- PRs that start `feature-`
- CI/CD checks that run additional testing (Python version matrix, for instance) and linting (ruff)

**Comment by canimus** [comment]
Hi @runkelcorey thanks for opening a good discussion. All items are pretty valid, happy to bring them on board.
I am sorry if the upgrade inflicted some morning-job-failure-pain, according the [pipy stats](https://pypistats.org/packages/cuallee), 99% of the download base was python 3.10, but understood that 1% also matters.

It came to my attention the idea of setting more clear development guidelines, and the use for instance of `pre-commit` to do `black` and `ruff` before actually opening the `PR` , some of this steps I think can be included in the Github Action running CI/CD. 

If you have some ideas, about the test matrix, or the steps to harness the feature branching, happy to implement them off course.
Again, thanks for the time and recommendations.

Regards,
H.
